crates/rust-analyzer/src/handlers.rs

```diff
@@ -985,6 +1003,33 @@ pub(crate) fn handle_code_lens_resolve(
             );
             Ok(CodeLens { range: code_lens.range, command: Some(cmd), data: None })
         }
+        Some(CodeLensResolveData::References(doc_position)) => {
+            let position = from_proto::file_position(&snap, doc_position.clone())?;
+            let locations = snap
+                .analysis
+                .find_all_refs(position, None)
```

Comment:
Hm, yeah, I am still not happy about factoring here -- way to many logic happens in the handlers. 

But it also is non-trivial to properly factor the logic given the two-phase requests. 

I think there's a better solution here, but I haven't looked in the details yet. I expect we want to use exactly the same pattern for code lenses as for assists.

Reply:
I just reuse existing find_all_refs method and convert the results to LSP types. And that's all.

>I think there's a better solution here, but I haven't looked in the details yet.

Perhaps we can (should) utilize the fact that Code Lens Request supports partial results. But it requires a lot of work and if you do not mind I'd prefer to leave it for another PR.